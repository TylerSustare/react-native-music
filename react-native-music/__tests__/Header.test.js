const React = require('React');
const ReactTestRenderer = require('react-test-renderer');
const Text = require('Text');
const { Header } = require('../components/Header');

describe('Header test', () => {
    it('renders correctly', (done) => {
        const instance = ReactTestRenderer.create(
            <Header />
        );

        expect(instance.toJSON()).toMatchSnapshot();
        done();
    });

    it('renders with header text', (done) => {
        const instance = ReactTestRenderer.create(
            <Header headerText={'test'} />
        );
        expect(instance.toJSON()).toMatchSnapshot();
        expect(instance.toJSON().children[0].children[0]).toBe('test');
        done();
    });
});