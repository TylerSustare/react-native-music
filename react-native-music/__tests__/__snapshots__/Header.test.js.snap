// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E8E8E8",
      "elevation": 2,
      "height": 60,
      "justifyContent": "center",
      "paddingTop": 15,
      "position": "relative",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</View>
`;

exports[`Header test renders with header text 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E8E8E8",
      "elevation": 2,
      "height": 60,
      "justifyContent": "center",
      "paddingTop": 15,
      "position": "relative",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    test
  </Text>
</View>
`;
